metalearn/metafeatures/resources.py

```diff
@@ -1,10 +1,64 @@
-"""
-Exposes all the package's bundled data files as file path strings.
-Needed to allow the data files to be successfully accessed across platforms 
-and install types.
-"""
+from typing import List
 
-import pkg_resources
+from metalearn.metafeatures.base import ResourceComputerMap, MetafeatureComputer, ResourceComputer
 
-METAFEATURE_CONFIG = pkg_resources.resource_filename('metalearn', 'metafeatures/metafeatures.json')
-METAFEATURES_JSON_SCHEMA = pkg_resources.resource_filename('metalearn', 'metafeatures/metafeatures_schema.json')
\ No newline at end of file
+from metalearn.metafeatures.decision_tree_metafeatures import resource_computers as dt_resources
+from metalearn.metafeatures.general_resource_computers import resource_computers as util_resources
+from metalearn.metafeatures.text_metafeatures import resource_computers as text_resources
+
+from metalearn.metafeatures.simple_metafeatures import metafeature_computers as simple_metafeatures
+from metalearn.metafeatures.statistical_metafeatures import metafeature_computers as statistical_metafeatures
+from metalearn.metafeatures.information_theoretic_metafeatures import metafeature_computers as info_theoretic_metafeatures
+from metalearn.metafeatures.landmarking_metafeatures import metafeature_computers as landmarking_metafeatures
+from metalearn.metafeatures.text_metafeatures import metafeature_computers as text_metafeatures
+from metalearn.metafeatures.decision_tree_metafeatures import metafeature_computers as dt_metafeatures
+
+
+def _get_metafeature_ids(metafeature_computers: List[MetafeatureComputer]) -> List[str]:
+    """Returns a list of all metafeature IDs found in `metafeature_computers`"""
+    metafeature_ids = set()
+    for computer in metafeature_computers:
+        for name in computer.returns:
+            if name in metafeature_ids:
+                raise ValueError("there is already a MetafeatureComputer that returns the {name} metafeature.")
+            metafeature_ids.add(name)
+    return list(metafeature_ids)
+
+
+resources_info = ResourceComputerMap()
+
+# Add all the ResourceComputers
+resources_info.add(dt_resources)
+resources_info.add(util_resources)
+resources_info.add(text_resources)
```

Comment:
we could make all the auxiliary resources their own `resources_info` mapping and just update the main one with the aux ones.

Reply:
I'm not fully sure what you mean here. I'd love to talk about it.